app/controllers/v0/sessions_controller.rb

```diff
@@ -35,7 +35,7 @@ def saml_callback
 
     def persist_session_and_user!
       @session = Session.new(user_attributes.slice(:uuid))
-      @current_user = User.find(@session.uuid) || User.new(user_attributes)
+      @current_user = User.find(@session.uuid) || UserMviDelegate.new(User.new(user_attributes)).create
```

Comment:
Interesting. And yes, I think it makes more sense to do it this way. `:after_initialize` is probably better suited for setting up defaults than fetching data. 👍